### DISCUSSION TITLE
PorousFlow: Fault mechanics

### DISCUSSION THREAD

**Discussion by wangshengwa** [idaholab/moose#22147] [Q&A Modules: General] [answered]
Hi experts,
I am simulating a fault in a rock aquifer with PorousFlow (by the way, thanks for the five start code). The geometry is complex and hydro-mechanical coupled.
I am first trying to reach mechanical equilibrium in steady state. But the geometry is complex, looks like the strains and displacements are big and therefore the different layers loose contact. See attached. Is there any way to "glue" the different layers? Or perhaps the finite strain approach is not correct?
![Screenshot from 2022-09-20 12-46-19](https://user-images.githubusercontent.com/113986975/191255394-d0a2fb40-d4d2-4dc8-8774-d507ca000850.png)

Cheers!
Wang

**Comment by WilkAndy** [comment, marked as answer]
If i understand your model correctly, this is a mesh problem, not a problem with your model.

Your deformed model exhibits two "fractures" (the grey lines in the above figure) that should not be there.  Those grey lines will not allow fluid to flow across them, nor will they transmit any mechanical force.  This is because they are actually empty space (there are no thin finite elements that sit inside those "fractures").

If the previous paragraph is correct, then your mesh is at fault.  You have created a mesh that contains a "left block" a "fault block" and a "right block" (shown in the above figure).  However, these blocks **do not share any nodes** except for at the very top of the picture (and probably the bottom too).  They must share nodes, and then your model will not generate those grey fractures.  You'll have to go back to your meshing software and ensure the blocks share nodes.

a

**Reply by wangshengwa** [reply]
Thanks!! This fixed the problem!!
